### ISSUE TITLE
Suggestion: Assign/remember application positions across virtual desktops

### ISSUE THREAD

**Issue by benz145** [mzomparelli/zVirtualDesktop#267]
I use five virtual desktops and always put the same applications on the same virtual desktop (ie: email and calendar on one, web browser on another, photoshop on another). 

I would love a way to launch all of my regular applications at once, and have them automatically go to the assigned virtual desktop. I know this is possible with some scripting magic like AutoHotkey, but that's not a particularly user-friendly solution for someone like myself.

Querying twitter, a few people said they had looked for or wished they had this functionality (including being willing to pay for it). Might be worth investigating as an additional feature of ZVD to draw in some more users for this use-case?

![image](https://user-images.githubusercontent.com/12601485/88227378-62659400-cc22-11ea-8652-2105d2b439e9.png)

![image](https://user-images.githubusercontent.com/12601485/88227395-67c2de80-cc22-11ea-8e66-74f372735641.png)

I did of course recommend ZVD to the latter response as it already includes hotkey switching.

**Comment by benz145** [comment]
Ha... seems I had already requested this feature a few years ago!

https://github.com/mzomparelli/zVirtualDesktop/issues/205

Totally forgot.

**Comment by mzomparelli** [comment]
Let me look at this again. If I'm understanding correctly, you simply want the on-demand ability to launch a predefined list of applications on a specified desktop.

**Comment by benz145** [comment]
Yes, with a single button it would be nice to launch a specific set of applications, each onto a specified virtual desktop. Ideally the user would be able to define not only a specific virtual desktop, but also a specific monitor, for each application (ie: for multi-monitor users).

**Comment by cpriest** [comment]
Ideally every application window would restore its-self back to the last desktop it was on.  Especially for browser windows.  

I noticed Chrome just started doing that a few months back, but I use Firefox.

While I'm not a paying user of this product (I have my own AHK script that does some of this), I would gladly pay for a license if it had this feature.  I imagine you'd have a swarm of new users if you got this to work.

Every time I reboot I'm spending time moving various browser windows back to the desktop they belong on.

**Comment by mzomparelli** [comment]
I can't think of a way to identify separate browser windows after a reboot. Every window gets a new window handle after a reboot and using the window text would not be a good idea for browser windows, but it might work for other windows.

If you have any ideas that could help with the actual implementation of this then please post it.

I think the best I can do is create a new hotkey action that will launch a list of apps, folders, urls, onto a specified desktop.

**Comment by cpriest** [comment]
When I was working on it w/ AHK I was storing a list of window titles and desktops, watching for window title changes and moving them as appropriately.  I went down the sinkhole of UIA in AHK which is tedious and never finished the project.

The titles would work well, assuming there was a blacklist functionality, like "don't move windows that are just named "Firefox" (because they are all named that at creation), but once the page loads the real window title will be there.

**Comment by mzomparelli** [comment]
Thanks for the feedback. I'll see what I can do with just window titles.

**Comment by Gideon-Felt** [comment]
I need this even if it only mostly works. But I can also just wait for windows 11.

**Comment by mzomparelli** [comment]
Is Windows 11 doing something with this? As far as I can tell Windows 11 is still just Windows 10. I actually haven't tried it yet though. Downloading the Windows 11 Insider Preview now.

I will try to get back to this feature request.

**Comment by mzomparelli** [comment]
I have tested out Windows 11 Insider and not much has changed with Virtual Desktops and zVD works with Windows 11. I will try to get busy on this feature request.
